src/main/java/org/squiddev/cobalt/LuaTable.java

```diff
@@ -688,10 +683,10 @@ public int prepSort() throws LuaError {
 	public boolean compare(LuaState state, int i, int j, LuaValue cmpfunc) throws LuaError, UnwindThrowable {
 		LuaValue a, b;
 
-		a = strengthen(array[i]);
-		b = strengthen(array[j]);
+		a = strengthen(rawget(i));
+		b = strengthen(rawget(j));
 
-		if (a.isNil() || b.isNil()) {
+		if (a.isNil() && b.isNil()) {
```

Comment:
How come `&&` rather than `||`?

Though looking at [PUC Lua's implementation](https://www.lua.org/source/5.1/ltablib.c.html), what we actually want is something like:

```java
if (!cmpfunc.isNil()) return OperationHelper.call(state, cmpfunc, a, b).toBoolean();
if (a.isNil() || b.isNil()) return false;
return OperationHelper.lt(state, a, b);
```

Reply:
Done